tensorrt_llm/_torch/disaggregation/native/transfer.py

```
@@ -684,6 +682,18 @@ def _deliver_kv_to_agent(self, write_meta: WriteMeta):
                 if all(t.status == TaskStatus.TRANSFERRED for t in session.kv_tasks):
                     session.transfer_end_time = tensorrt_llm.bindings.global_steady_clock_now()
 
+        # Perf logging only after completion is signalled: the CSV write can
+        # stall on shared storage and must never delay the ctx block-all
+        # waiter (https://nvbugs/6669114).
+        try:
+            ri = self._registrar.self_rank_info
+            task.print_perf_info(write_meta.peer_rank, ri.instance_name, ri.instance_rank)
+        except Exception as e:  # perf is best-effort; never block completion
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)